File: it/test-versions/expected/suite/java/bar/_1_0_0/Other.java
package bar._1_0_0;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Objects;
import java.util.Optional;

public class Other {
  private final String name;

  @JsonCreator
  public Other(
    @JsonProperty("name") final String name
  ) {
    Objects.requireNonNull(name, "name");
    this.name = name;
  }

  public String getName() {
    return this.name;
  }

  @Override
  public int hashCode() {
    int result = 1;
    result = result * 31 + this.name.hashCode();
    return result;
  }

  @Override
  public boolean equals(final Object other) {
    if (other == null) {
      return false;
    }

    if (!(other instanceof Other)) {
      return false;
    }

    @SuppressWarnings("unchecked")
    final Other o = (Other) other;

    if (!this.name.equals(o.name)) {
      return false;
    }

    return true;
  }

  @Override
  public String toString() {
    final StringBuilder b = new StringBuilder();

    b.append("Other");
    b.append("(");
    b.append("name=");
    b.append(this.name.toString());
    b.append(")");

    return b.toString();
  }

  public static class Builder {
    private Optional<String> name = Optional.empty();

    public Builder name(final String name) {
      this.name = Optional.of(name);
      return this;
    }

    public Other build() {
      final String name = this.name.orElseThrow(() -> new RuntimeException("name: is required"));

      return new Other(name);
    }
  }
}

File: it/test-versions/expected/suite/csharp/Bar/_1_0_0/Other.cs
using Newtonsoft.Json;
using System;
using System.Text;

namespace Bar._1_0_0 {
  [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
  public class Other {
    [JsonProperty("name", Required = Required.DisallowNull)]
    public String name {
      get;
    }

    [JsonConstructor]
    public Other(
      [JsonProperty("name", Required = Required.DisallowNull)] String name
    ) {
      this.name = name;
    }

    public override Int32 GetHashCode() {
      Int32 result = 1;
      result = result * 31 + this.name.GetHashCode();
      return result;
    }

    public override Boolean Equals(Object other) {
      Other o = other as Other;

      if (o == null) {
        return false;
      }

      if (!this.name.Equals(o.name)) {
        return false;
      }

      return true;
    }

    public override String ToString() {
      StringBuilder b = new StringBuilder();

      b.Append("Other");
      b.Append("(");
      b.Append("name=");
      b.Append(this.name);
      b.Append(")");

      return b.ToString();
    }
  }
}
